jQuery/ajax.js

```diff
@@ -6,6 +6,18 @@ function ajax(citizens) {
   populateSelectWithOptions("#character-select", characters);
   populateSelectWithOptions("#hometown-select", hometowns);
   populateSelectWithOptions("#size-select", sizes);
+
+  var selectHandler = function(event) {
+    var params = gatherParams();
+    var filteredCitizens = filter(params, event.data.citizens);
+    redrawContent(filteredCitizens);
+  };
+
+  $('#character-select').on('change', { citizens: citizens }, selectHandler);
+  $('#hometown-select').on('change', { citizens: citizens }, selectHandler);
+  $('#size-select').on('change', { citizens: citizens }, selectHandler);
+
+  selectHandler({data: { citizens: citizens }});
```

Comment:
Well. You're not really able to prepopulate it without making an ajax call and manually "redrawing" . A lot of frameworks do this. 

This is just kind of slow because the DOM is the bottleneck. there are only so many times per second you can make changes to the DOM.